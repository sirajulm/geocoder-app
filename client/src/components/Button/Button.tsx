import React, { SyntheticEvent } from 'react'
import styled from 'styled-components';

/**
 * Button can be of 2 kind primary and secondary with styling changes
 */
interface ButtonProps {
    onClick: (event: SyntheticEvent) => void;
    kind: 'primary' | 'secondary'
}


const StyledButton = styled.button`
    ${(props: ButtonProps) => (props.kind === 'primary') ?
        {
            'background-color': '#165fa4',
            'border': 0,
            'color': '#ffffff'
        } : {
            'background-color': '#ffffff',
            'border': '1px solid #dadada',
            'color': '#000000'
        }
    }
    width: 100px;
    height: 40px;
    margin: 15px;
    border-radius: 5px;
    font-size: 16px;
    padding:0 15px;
    outline: none;
`;

export default StyledButton;