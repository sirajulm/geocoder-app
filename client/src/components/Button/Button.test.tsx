import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import 'jest-styled-components'


import Button from './Button';


describe('Button', () => {
    it('should render', () => {
        const component = renderer.create(<Button />).toJSON()
        expect(component).toMatchSnapshot()
        expect(component).toHaveStyleRule('background-color', '#ffffff')
    });

    it('should render primary', () => {
        const component = renderer.create(<Button kind="primary" />).toJSON()
        expect(component).toMatchSnapshot()
        expect(component).toHaveStyleRule('background-color', '#165fa4')
    })

    it('should render secondary', () => {
        const component = renderer.create(<Button kind="secondary" />).toJSON()
        expect(component).toMatchSnapshot()
        expect(component).toHaveStyleRule('background-color', '#ffffff')
    })
    it('should render label', () => {
        const label = 'Test';
        const component = mount(<Button>{label}</Button>)
        const text = component.find('button').text()
        expect(text).toEqual(label)
    })
})
