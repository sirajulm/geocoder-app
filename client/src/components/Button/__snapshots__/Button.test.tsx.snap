// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
.c0 {
  background-color: #ffffff;
  border: 1px solid #dadada;
  color: #000000;
  width: 100px;
  height: 40px;
  margin: 15px;
  border-radius: 5px;
  font-size: 16px;
  padding: 0 15px;
  outline: none;
}

<button
  className="c0"
/>
`;

exports[`Button should render primary 1`] = `
.c0 {
  background-color: #165fa4;
  border: 0;
  color: #ffffff;
  width: 100px;
  height: 40px;
  margin: 15px;
  border-radius: 5px;
  font-size: 16px;
  padding: 0 15px;
  outline: none;
}

<button
  className="c0"
  kind="primary"
/>
`;

exports[`Button should render secondary 1`] = `
.c0 {
  background-color: #ffffff;
  border: 1px solid #dadada;
  color: #000000;
  width: 100px;
  height: 40px;
  margin: 15px;
  border-radius: 5px;
  font-size: 16px;
  padding: 0 15px;
  outline: none;
}

<button
  className="c0"
  kind="secondary"
/>
`;
